Quarkus 1.13.0.CR1 dev mode fails with build exception on Apple Silicon machine

## Describe the bug
Quarkus dev mode fails with 1.13.0.CR1 on arm64 and x86_64 Java when using MacBook with M1 processor. 

I verified it works on my Intel machine (although using java 15 not java 11). On the M1 machine package jar and native build also run fine. 

Quarkus dev mode running 1.12.x works fine on M1 Mac.

## To Reproduce

Download quarkus-quickstarts. Go to getting started and adjust pom file to use 1.13.0.CR1. 

Steps to reproduce the behavior:
1. Download quarkus-quickstarts
2. In getting-started adjust pom to use `1.13.0.CR1`
3. run `mvn compile quarkus:dev`

## Environment (please complete the following information):

### Output of `uname -a` or `ver`
`Darwin jc-polytomic.lan 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:06:51 PST 2021; root:xnu-7195.81.3~1/RELEASE_ARM64_T8101 arm64`

### Output of `java -version`

arm64 from brew:
```
❯ java -version
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9)
OpenJDK 64-Bit Server VM (build 11.0.10+9, mixed mode)
```

x86_64 from AdoptOpenJDK
```
❯ JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home java -version
openjdk version "11.0.10" 2021-01-19
OpenJDK Runtime Environment (build 11.0.10+9)
OpenJDK 64-Bit Server VM (build 11.0.10+9, mixed mode)
```

Installed Javas:
```
❯ /usr/libexec/java_home -V
Matching Java Virtual Machines (3):
    11.0.10 (arm64) "UNDEFINED" - "OpenJDK 11.0.10" /opt/homebrew/Cellar/openjdk@11/11.0.10/libexec/openjdk.jdk/Contents/Home
    11.0.10 (x86_64) "GraalVM Community" - "GraalVM CE 21.0.0" /Library/Java/JavaVirtualMachines/graalvm-ce-java11-21.0.0/Contents/Home
    11.0.10 (x86_64) "AdoptOpenJDK" - "AdoptOpenJDK 11" /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
/opt/homebrew/Cellar/openjdk@11/11.0.10/libexec/openjdk.jdk/Contents/Home
```

## Additional context

```
~/projects/quarkus-quickstarts/getting-started main* 12s                                                                               09:49:12
❯ git -P diff
diff --git a/getting-started/pom.xml b/getting-started/pom.xml
index f93a1c79..a7a78ccb 100644
--- a/getting-started/pom.xml
+++ b/getting-started/pom.xml
@@ -7,10 +7,10 @@
     <version>1.0.0-SNAPSHOT</version>

     <properties>
-        <quarkus-plugin.version>1.12.2.Final</quarkus-plugin.version>
+        <quarkus-plugin.version>1.13.0.CR1</quarkus-plugin.version>
         <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
         <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-        <quarkus.platform.version>1.12.2.Final</quarkus.platform.version>
+        <quarkus.platform.version>1.13.0.CR1</quarkus.platform.version>
         <compiler-plugin.version>3.8.1</compiler-plugin.version>
         <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

~/projects/quarkus-quickstarts/getting-started main*                                                                                   09:49:32
❯ JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home ./mvnw clean compile quarkus:dev
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.acme:getting-started >----------------------
[INFO] Building getting-started 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ getting-started ---
[INFO] Deleting /Users/james/projects/quarkus-quickstarts/getting-started/target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ getting-started ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ getting-started ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /Users/james/projects/quarkus-quickstarts/getting-started/target/classes
[INFO]
[INFO] --- quarkus-maven-plugin:1.13.0.CR1:dev (default-cli) @ getting-started ---
Listening for transport dt_socket at address: 5005
2021-03-18 09:49:42,549 INFO  [org.jbo.threads] (main) JBoss Threads version 3.2.0.Final
2021-03-18 09:49:43,531 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.ide.IdeProcessor#detectRunningIdeProcesses threw an exception: java.lang.RuntimeException: Unable to determine running processes
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:196)
	at io.quarkus.deployment.ide.IdeProcessor.detectRunningIdeProcesses(IdeProcessor.java:138)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:191)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180)
	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:699)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:181)
	... 11 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:183)
	... 22 more
Caused by: java.lang.RuntimeException: Input/output error
	at java.base/java.lang.ProcessHandleImpl$Info.info0(Native Method)
	at java.base/java.lang.ProcessHandleImpl$Info.info(ProcessHandleImpl.java:578)
	at java.base/java.lang.ProcessHandleImpl.info(ProcessHandleImpl.java:490)
	... 26 more

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:396)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:269)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:66)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:79)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:378)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:127)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:84)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:144)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.ide.IdeProcessor#detectRunningIdeProcesses threw an exception: java.lang.RuntimeException: Unable to determine running processes
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:196)
	at io.quarkus.deployment.ide.IdeProcessor.detectRunningIdeProcesses(IdeProcessor.java:138)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:191)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180)
	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:699)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:181)
	... 11 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:183)
	... 22 more
Caused by: java.lang.RuntimeException: Input/output error
	at java.base/java.lang.ProcessHandleImpl$Info.info0(Native Method)
	at java.base/java.lang.ProcessHandleImpl$Info.info(ProcessHandleImpl.java:578)
	at java.base/java.lang.ProcessHandleImpl.info(ProcessHandleImpl.java:490)
	... 26 more

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:153)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:394)
	... 9 more
Caused by: java.lang.RuntimeException: Unable to determine running processes
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:196)
	at io.quarkus.deployment.ide.IdeProcessor.detectRunningIdeProcesses(IdeProcessor.java:138)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:920)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2415)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:191)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.IntPipeline$1$1.accept(IntPipeline.java:180)
	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:699)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.runningProcesses(IdeProcessor.java:181)
	... 11 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ide.IdeProcessor$ProcessUtil.lambda$runningProcesses$0(IdeProcessor.java:183)
	... 22 more
Caused by: java.lang.RuntimeException: Input/output error
	at java.base/java.lang.ProcessHandleImpl$Info.info0(Native Method)
	at java.base/java.lang.ProcessHandleImpl$Info.info(ProcessHandleImpl.java:578)
	at java.base/java.lang.ProcessHandleImpl.info(ProcessHandleImpl.java:490)
	... 26 more

2021-03-18 09:49:43,531 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
^C%
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java
@@ -17,6 +17,8 @@ import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.stream.Stream;
 
+import org.jboss.logging.Logger;
+
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.pkg.builditem.BuildSystemTargetBuildItem;
@@ -24,6 +26,8 @@ import io.quarkus.dev.spi.DevModeType;
 import io.quarkus.runtime.util.JavaVersionUtil;
 
 public class IdeProcessor {
+
+    private static final Logger log = Logger.getLogger(IdeProcessor.class);
 
     private static Map<String, List<Ide>> IDE_MARKER_FILES = new HashMap<>();
     private static Map<Predicate<ProcessInfo>, Ide> IDE_PROCESSES = new HashMap<>();
@@ -135,7 +139,13 @@ public class IdeProcessor {
             return null;
         }
         Set<Ide> result = new HashSet<>(4);
-        List<ProcessInfo> processInfos = ProcessUtil.runningProcesses();
+        List<ProcessInfo> processInfos = Collections.emptyList();
+        try {
+            processInfos = ProcessUtil.runningProcesses();
+        } catch (Exception e) {
+            // this shouldn't be a terminal failure, so just log it to the console
+            log.warn(e.getMessage());
+        }
         for (ProcessInfo processInfo : processInfos) {
             for (Map.Entry<Predicate<ProcessInfo>, Ide> entry : IDE_PROCESSES.entrySet()) {
                 if (entry.getKey().test(processInfo)) {
@@ -193,7 +203,7 @@ public class IdeProcessor {
                 });
                 return result;
             } catch (Exception e) {
-                throw new RuntimeException("Unable to determine running processes", e);
+                throw new RuntimeException("Unable to determine running IDE processes", e);
             }
         }
     }

